packages/eas-cli/src/analytics/AnalyticsManager.ts

```diff
@@ -0,0 +1,230 @@
+  }
+
+  const persistedDeviceId = await UserSettings.getAsync(
+    USER_SETTINGS_KEY_ANALYTICS_DEVICE_ID,
+    null
+  );
+  const deviceId = persistedDeviceId ?? uuidv4();
+  if (!persistedDeviceId) {
+    await UserSettings.setAsync(USER_SETTINGS_KEY_ANALYTICS_DEVICE_ID, deviceId);
+  }
+  return new AnalyticsManager(deviceId);
+}
+
+class NoOpAnalyticsManager implements IAnalayticsManagerWithOrchestration {
+  logEvent(): void {}
+  setActor(): void {}
+  async flushAsync(): Promise<void> {}
+}
+
+const AnalyticsManagerConfig =
+  process.env.EXPO_STAGING || process.env.EXPO_LOCAL
+    ? {
+        // staging environment
+        rudderstackWriteKey: '1wpX20Da4ltFGSXbPFYUL00Chb7',
+        rudderstackDataPlaneURL: 'https://cdp.expo.dev',
+      }
+    : {
+        // prod environment
+        rudderstackWriteKey: '1wpXLFxmujq86etH6G6cc90hPcC',
+        rudderstackDataPlaneURL: 'https://cdp.expo.dev',
+      };
+
+class AnalyticsManager implements IAnalayticsManagerWithOrchestration {
+  private readonly rudderstackClient = new RudderAnalytics(
+    AnalyticsManagerConfig.rudderstackWriteKey,
+    new URL('/v1/batch', AnalyticsManagerConfig.rudderstackDataPlaneURL).toString(),
+    {
+      flushInterval: 300,
+    }
+  );
+
+  private identifiedActor: Actor | null = null;
+
+  constructor(private readonly persistentDeviceId: string) {
+    // identify once with just anonymous ID. Once the actor is fetched, re-indentify with
+    // both so that they can be associated
+    this.rudderstackClient.identify({
+      anonymousId: persistentDeviceId,
+    });
+  }
+
+  public setActor(actor: Actor): void {
+    if (this.identifiedActor) {
+      return;
+    }
+
+    this.rudderstackClient.identify({
+      userId: actor.id,
+      anonymousId: this.persistentDeviceId,
+      traits: {
+        username: getActorDisplayName(actor),
+        user_id: actor.id,
+        user_type: actor.__typename,
+      },
+    });
+    this.identifiedActor = actor;
+  }
+
+  public logEvent(name: AnalyticsEvent, properties: Record<string, any>): void {
+    const userId = this.identifiedActor?.id;
+    const deviceId = this.persistentDeviceId;
+    const commonEventProperties = { source_version: easCliVersion, source: 'eas cli' };
+    const identity = { userId: userId ?? undefined, anonymousId: deviceId ?? uuidv4() };
+    this.rudderstackClient.track({
+      event: name,
+      properties: { ...properties, ...commonEventProperties },
+      ...identity,
+      context: this.getRudderStackContext(),
+    });
+  }
+
+  public async flushAsync(): Promise<void> {
+    await this.rudderstackClient.flush();
+  }
+
+  private getRudderStackContext(): Record<string, any> {
+    const platform = PLATFORM_TO_ANALYTICS_PLATFORM[os.platform()] || os.platform();
+    return {
+      os: { name: platform, version: os.release() },
+      device: { type: platform, model: platform },
+      app: { name: 'eas cli', version: easCliVersion ?? undefined },
+    };
+  }
+}
```

Comment:
Github won't let me comment on the file below (`NoOpAnalyticsManager.ts`) - the file is empty and not referenced anywhere. Is this intentional?